apps/mux/src/config.tsx

```diff
@@ -0,0 +1,109 @@
+    super(props);
+    this.state = { parameters: {} };
+
+    // `sdk.app` exposes all app-related methods.
+    this.app = this.props.sdk.app;
+
+    // `onConfigure` allows to configure a callback to be
+    // invoked when a user attempts to install the app or update
+    // its configuration.
+    this.app.onConfigure(() => this.onConfigure());
+  }
+
+  async componentDidMount () {
+    // Get current parameters of the app.
+    const parameters = await this.app.getParameters();
+
+    this.setState(
+      // If the app is not installed, `parameters` will be `null`.
+      // We default to an empty object in this case.
+      { parameters: parameters || {} },
+      () => {
+        // Once preparation has finished, call `setReady` to hide
+        // the loading screen and present the app to a user.
+        this.app.setReady()
+      }
+    );
+  }
+
+  // Renders the UI of the app.
+  render () {
+    const { parameters: { muxAccessTokenId, muxAccessTokenSecret } } = this.state;
+
+    return (
+      <React.Fragment>
+        <div className='form-wrapper'>
+          <Form id="app-config" spacing="default">
+            <Heading>Mux Video Uploader</Heading>
+            <Note noteType="primary" title="About the app">
+              A Contentful UI app that makes it simple to add beautiful streaming via <a href='https://mux.com'>Mux</a> to your Contentful project. Install the extension, add the component to your content model, and you're good to go! 🙌
+            </Note>
+            <TextField
+              required
+              name="mux-access-token"
+              id="mux-access-token"
+              labelText="Mux access token"
+              value={muxAccessTokenId || ''}
+              onChange={e => this.setState({ parameters: { muxAccessTokenId: e.target.value, muxAccessTokenSecret } })}
+            />
+            <TextField
+              required
+              name="mux-token-secret"
+              id="mux-token-secret"
+              labelText="Mux token secret"
+              value={muxAccessTokenSecret || ''}
+              onChange={e => this.setState({ parameters: { muxAccessTokenId, muxAccessTokenSecret: e.target.value } })}
+            />
+          </Form>
+        </div>
+      </React.Fragment>
+    );
+  }
+
+  // This method will be called when a user clicks on "Install"
+  // or "Save" in the configuration screen.
+  //
+  // We want to do two things in here:
+  // 1. Persist selected animal as a parameter
+  // 2. Put the app to sidebars of all Content Types
+  async onConfigure () {
+    // Get IDs of all content types in an environment.
+    const { items: contentTypes } = await this.props.sdk.space.getContentTypes();
+    const contentTypeIds = contentTypes.map(ct => ct.sys.id)
+
+    // Return value of `onConfigure` is used to install
+    // or update the configuration.
+    return {
+      // Parameters to be persisted as the app configuration.
+      parameters: this.state.parameters,
```

Comment:
It looks like you aren't actually checking to see if an ID or token were provided. The app cannot function without either.

I'd suggest doing some data validation here to check that the two values you are expecting actually exist before allowing the user to save this configuration.

When you `return false` in this function, we cancel the installation/save process from happening in the webapp. For example, some code might look like:
```js
if (!this.state.parameters.muxAccessTokenSecret.trim()) {
  this.props.sdk.notify.error('You must provide an access token!');
  return false;
}

Reply:
Done

![mux-contentful](https://user-images.githubusercontent.com/764988/81205032-fbb7cf00-8f7e-11ea-80d5-3182a58e1538.gif)